### PR TITLE
EICNET-942: As a GO I can delete my group when it's in pending state

### PR DESCRIPTION
**Pull request by 4hardfire**
### Changes

- New group permission helper methods added to eic_groups.helper service;
- Add/remove **group delete** permission from group owner when moderation state is **pending**.

### Tests

- [ ] As a trusted user, create a new group;
- [ ] Go to the group edit page and check if the delete link is available; Also, validate if the user has access to -> `groups/<group-title>/delete`
- [ ] As a SA/SCM update the group status to "Draft";
- [ ] As a trusted user, go to the group edit page and check if the delete link is not visible; Also, validate if the access is denied in -> `groups/<group-title>/delete`.

### Todo

- [ ] The delete button in the group header block is not yet available and needs to be added after we merge #336

### CODE REVIEW

File: lib/modules/eic_groups/src/Hooks/EntityOperations.php [hunk 1]
```diff
@@ -103,6 +123,25 @@ public function groupUpdate(GroupInterface $entity) {
...
+  }
+
+  /**
+   * Implements hook_group_permission_insert().
+   */
+  public function groupPermissionInsert(GroupPermissionInterface $group_permissions) {
+    $group = $group_permissions->getGroup();
+    // Adds or removes delete group permission from group owner based on the
+    // group moderation state.
+    if ($group->get('moderation_state')->value === 'pending') {
```
Comment (lramarojaona) [inline lib/modules/eic_groups/src/Hooks/EntityOperations.php:137]: Maybe introduce constants for the moderation states?

File: lib/modules/eic_groups/src/Hooks/EntityOperations.php [hunk 2]
```diff
@@ -263,4 +302,28 @@ protected function createGroupAboutPageMenuLink(EntityInterface $group) {
...
+   */
+  protected function updateGroupOwnerPermissions(GroupInterface $entity) {
+    /** @var \Drupal\group_permissions\Entity\GroupPermissionInterface $group_permissions */
+    $group_permissions = $this->groupPermissionsManager->loadByGroup($entity);
+
+    // Get moderation states.
+    $old_moderation_state = $entity->original->get('moderation_state')->value;
+    $new_moderation_state = $entity->get('moderation_state')->value;
+
+    if ($old_moderation_state !== $new_moderation_state && $new_moderation_state === 'pending') {
```
Comment (lramarojaona) [inline lib/modules/eic_groups/src/Hooks/EntityOperations.php:319]: Could we have a small comment to explain this condition?

File: lib/modules/eic_groups/src/Hooks/EntityOperations.php [hunk 3]
```diff
@@ -263,4 +302,28 @@ protected function createGroupAboutPageMenuLink(EntityInterface $group) {
...
+      $this->eicGroupsHelper->addRolePermissionsToGroup($group_permissions, EICGroupsHelper::GROUP_OWNER_ROLE, ['delete group']);
+      $this->eicGroupsHelper->saveGroupPermissions($group_permissions);
+    }
+    elseif ($old_moderation_state !== $new_moderation_state && $new_moderation_state !== 'pending') {
```
Comment (lramarojaona) [inline lib/modules/eic_groups/src/Hooks/EntityOperations.php:323]: Could we have a small comment to explain this condition?

**Review by lramarojaona — CHANGES_REQUESTED** [review]

File: lib/modules/eic_groups/src/Hooks/EntityOperations.php (hunk 2 above)
Comment (4hardfire) [reply lib/modules/eic_groups/src/Hooks/EntityOperations.php]: Done. Can you have a look?

File: lib/modules/eic_groups/src/Hooks/EntityOperations.php (hunk 3 above)
Comment (4hardfire) [reply lib/modules/eic_groups/src/Hooks/EntityOperations.php]: Done. Can you have a look?

File: lib/modules/eic_groups/src/Hooks/EntityOperations.php (hunk 1 above)
Comment (4hardfire) [reply lib/modules/eic_groups/src/Hooks/EntityOperations.php]: Done.

**Pull request merged**